internal/webhook/inferenceservice/mutating.go

```diff
@@ -127,7 +155,7 @@ func (w *ConnectionWebhook) Handle(ctx context.Context, req admission.Request) a
 			return admission.Allowed(fmt.Sprintf("No connection injection performed for %s in namespace %s", req.Kind.Kind, req.Namespace))
 
 		case webhookutils.ConnectionActionRemove:
-			// Perform cleanup when annotation is removed
+			// Perform cleanup when annotation is removed, we do not delete SA but only remove injection part
```

Comment:
Do we ever want to delete the SA? What if not just the annotation is removed but the entire `InferenceService` object is deleted - would we want to delete the SA then?